itou/employee_record/models.py

```diff
@@ -469,13 +469,11 @@ def asp_siae_type(self):
     @staticmethod
     def siret_from_asp_source(siae):
         """
-        Fetch SIRET number of ASP source structure ("mother" SIAE)
+        Fetch SIRET number of authoritative SIAE from ASP source
         """
-        if siae.source != Company.SOURCE_ASP:
-            main_siae = Company.objects.get(convention=siae.convention, source=Company.SOURCE_ASP)
-            return main_siae.siret
-
-        return siae.siret
+        if siae.canonical_company.source == Company.SOURCE_ASP:
+            return siae.canonical_company.siret
```

Comment:
Pour éviter d'aller chercher la convention et les entreprises associées alors qu'il y a de forte chance qu'on soit déjà sur la bonne :)
```suggestion
        if siae.source == Company.SOURCE_ASP:
            return siae.siret
        if siae.canonical_company.source == Company.SOURCE_ASP:
            return siae.canonical_company.siret
```

Reply:
@rsebille En fait si `siae.source == Company.SOURCE_ASP`, `canonical_company` renverra trivialement `self` sans chercher ni convention ni entreprises associées car on entre pas dans la clause `if self.convention_id and self.source == self.SOURCE_USER_CREATED:`. Du coup pas besoin non ?